tests/src/test/scala/com/scylladb/migrator/alternator/DynamoDBInputFormatTest.scala

```diff
@@ -0,0 +1,79 @@
+package com.scylladb.migrator.alternator
+
+import com.amazonaws.services.dynamodbv2.model.{BillingMode, BillingModeSummary, ProvisionedThroughputDescription, TableDescription}
+import com.scylladb.migrator.readers.DynamoDB
+import org.apache.spark.sql.SparkSession
+
+class DynamoDBInputFormatTest extends munit.FunSuite {
+
+  val spark = SparkSession.builder().appName("test").master("local[*]").getOrCreate()
+
+  val GB: Long = 1024 * 1024 * 1024
+
+  test("no configured scanSegments in on-demand billing mode") {
+    checkPartitions(10)(tableSizeBytes = 1 * GB, tableProvisionedThroughput = None)
+  }
+
+  test("no configured scanSegments in on-demand billing mode and table size is 100 GB") {
+    checkPartitions(1024)(tableSizeBytes = 100 * GB, tableProvisionedThroughput = None)
+  }
+
+  test("no configured scanSegments in provisioned billing mode") {
+    checkPartitions(10)(tableSizeBytes = 1 * GB, tableProvisionedThroughput = Some((25, 25)))
+  }
```

Comment:
When the billing mode is provisioned, we still get 10 chunks in the current algorithm because there is no point in using a higher number of smaller chunks, AFAIU.